pkg/component/etcd/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,175 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type component struct {
+	client           client.Client
+	namespace        string
+	values           *Values
+	controllerConfig *rest.Config
+}
+
+const (
+	ownedByAnnotationKey   = "gardener.cloud/owned-by"
+	ownerTypeAnnotationKey = "gardener.cloud/owner-type"
+)
+
+// New creates a new poddisruptionbudget deployer instance.
+func New(c client.Client, namespace string, values *Values, controllerConfig *rest.Config) gardenercomponent.Deployer {
+	return &component{
+		client:           c,
+		namespace:        namespace,
+		values:           values,
+		controllerConfig: controllerConfig,
+	}
+}
+
+// getOwnerReferences returns owner references for the PDB object
+func getOwnerReferences(val *Values) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		{
+			APIVersion:         druidv1alpha1.GroupVersion.String(),
+			Kind:               "Etcd",
+			Name:               val.EtcdName,
+			UID:                val.EtcdUID,
+			Controller:         pointer.BoolPtr(true),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+		},
+	}
+}
+
+// getLabels returns the labels for the PDB object
+func getLabels(val *Values) map[string]string {
+	labels := map[string]string{
+		"instance": val.EtcdName,
+	}
+
+	for k, v := range val.Labels {
+		labels[k] = v
+	}
+
+	return labels
+}
+
+// getAnnotations returns the annotations for the PDB object
+func getAnnotations(val *Values) map[string]string {
+	annotations := map[string]string{
+		ownedByAnnotationKey:   fmt.Sprintf("%s/%s", val.EtcdNameSpace, val.EtcdName),
+		ownerTypeAnnotationKey: "etcd",
+	}
+	return annotations
+}
+
+// emptyPodDisruptionBudget returns an empty PDB object with only the name and namespace as part of the object meta
+func (c *component) emptyPodDisruptionBudget(name string, k8sversion *semver.Version) interface{} {
+	fmt.Println("In PDB emptyPDB")
+	if k8sutils.ConstraintK8sGreaterEqual121.Check(k8sversion) {
+		return &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: c.namespace,
+			},
+		}
+	} else {
+		return &policyv1beta1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: c.namespace,
+			},
+		}
+	}
+}
+
+func (c *component) Deploy(ctx context.Context) error {
+	fmt.Println("In PDB deploy")
+	k8sversion, _ := utils.GetTargetK8sVersion(c.controllerConfig)
+	pdb := c.emptyPodDisruptionBudget(c.values.EtcdName, k8sversion)
+
+	if err := c.syncPodDisruptionBudget(ctx, pdb); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *component) syncPodDisruptionBudget(ctx context.Context, pdb interface{}) error {
+	switch pdb := pdb.(type) {
+	case *policyv1.PodDisruptionBudget:
+		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, pdb, func() error {
```

Comment:
Will `OperationResult` which is the ignored argument contain additional information when `GetAndCreateOrMergePatch` returns an error?

Reply:
`OperationResult` is just a string that indicates the executed operation (ref: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#OperationResult)

It's just "unchanged" in case of an error, so no, no additional in case of an error

Reply:
ok